front/src/app/todo-list/components/active-task-list/active-task-list.component.ts

```diff
@@ -10,35 +11,60 @@ import { ApiService } from '../../../shared/api-service/api.service';
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class ActiveTaskListComponent implements OnDestroy, OnInit {
-  @Input() listId$: Observable<string> | undefined;
-
+  isRenameTaskListVisible: boolean = false
+ 
   taskList: TaskList | undefined;
 
+  activeTaskList: TaskGroup[] | undefined;
+
   isNewTaskInputVisible: boolean = false;
 
-  activeListId: string = '';
+  activeListId: string | undefined;
 
   private destroy$ = new Subject<void>();
 
-  constructor(public apiService: ApiService, private changeDetectorRef: ChangeDetectorRef) {
+  constructor(private apiService: ApiService, private taskListManager: TaskListManagerService, private changeDetectorRef: ChangeDetectorRef) {
   } 
 
   ngOnInit(): void {
-    this.listId$?.pipe(
+    this.taskListManager.listId$.pipe(
+      takeUntil(this.destroy$)
+    )
+    .subscribe(listId => {
+      this.activeListId = listId;
+    })
+
+    this.taskListManager.allTaskLists
+      .pipe(
+        filter<TaskGroupsList | undefined, TaskGroupsList>((data): data is TaskGroupsList => !!data),
+        takeUntil(this.destroy$)
+      )
+      .subscribe(data => {
+        if (data.lists) {
+          this.getActiveTaskList(this.activeListId);
+          this.changeDetectorRef.markForCheck();
+        }

```

Comment:
Change `this.activeListId` and `this.activeTaskList` to Observables

For `this.activeTaskList` consider using RxJs `combineLatest` for getting result based on `this.taskListManager.allTaskLists` and `this.taskListManager.listId$`